runtime/lua/vim/shared.lua

```diff
@@ -91,12 +95,12 @@ end
 ---
 --- @param s string String to split
 --- @param sep string Separator or pattern
---- @param opts (table|nil) Keyword arguments |kwargs|:
+--- @param opts? vim.gsplit.Opts (table) Keyword arguments |kwargs|:
 ---       - plain: (boolean) Use `sep` literally (as in string.find).
 ---       - trimempty: (boolean) Discard empty segments at start and end of the sequence.
```

Comment:
These docstrings are on the `@class vim.gsplit.Opts`, so why do we still need them here? Is this a current limitation of the lua2dox script?

Reply:
Lua2dox currently doesn't render `@class`, but I want to add this capability soonish. Want to address all the remaining warnings first.

Reply:
On that note, do you have any thoughts on incorporating LuaLS into our documentation flow? LuaLS has the ability to parse a file and render a JSON of its structure for documentation purposes.

Pro: should simplify scripts significantly whilst hopefully making them more robust.
Con: LuaLS becomes a build dependency.

On the otherhand maintaining our own LuaCATS parser:

Pro: gives us much greater flexibility, e.g. supporting custom annotations like `@nodoc`.
Con: more scripts at home.

For now, I think our scripts are fairly maintainable (ignoring the whole doxygen middle-man), so I'm happy doing our own thing for the time being.

Reply:
> Con: LuaLS becomes a build dependency.

We can make compromises for "developer/CI requirements" compared to "build requirements". If LuaLS has a programmatic mode that allows us to eliminate lua2dox then it's probably worth it.

But we shouldn't use LuaLS to feed stuff into `gen_vimdoc.py`, that would be throwaway work. Instead we would want to couple that with a new script (similar to `gen_help_html.lua`) that generates docs directly from the (non-doxygen) structure.

Reply:
> Con: LuaLS becomes a build dependency.

That's a _very_ big con in my view, given how unpredictable upstream development is.